### DISCUSSION TITLE
New extension for listings of cross-references

### DISCUSSION THREAD

**Discussion by tobydriscoll** [quarto-dev/quarto-cli#5745] [Show and Tell]
### Description

In a book with lots of examples, definitions, etc. scattered across many chapters, it can be handy for the reader to have a single list or table of all such items. I've written an [extension](https://www.github.com/tobydriscoll/crossref-listings) to help automate this process.

It's pretty clunky. I'm new to Lua and Pandoc coding, and Pandoc isn't natively set up to work across multiple files. Someone who really knows Quatro could have done it faster and better, I'm sure. But it works, and I know students really appreciated a one-stop summary of examples this past semester.

To get started, use `quarto add tobydriscoll/crossref-listings` in your project directory. The extension directory has an explanation that is also a demo.

**Comment by mcanouil** [comment]
You might want to add your extension to Quarto website, see <https://quarto.org/docs/extensions/>.

**Comment by cscheid** [comment]
This looks good, thanks for sharing! We have ticket #2138 for precisely this issue in quarto, it's great to see that it's doable as an extension.
